packages/react-instantsearch-core/src/connectors/connectRefinementList.js

```diff
@@ -122,8 +122,11 @@ export default createConnector({
     metadata,
     searchForFacetValuesResults
   ) {
-    const { attribute, searchable } = props;
-    const results = getResults(searchResults, { ais: props.contextValue });
+    const { attribute, searchable, indexContextValue } = props;
+    const results = getResults(searchResults, {
+      ais: props.contextValue,
+      multiIndexContext: props.indexContextValue,
```

Comment:
Since we destruct the value above we can re-use it. We can apply the change in the all function.